acceptance/mockAgent/daemon.go

```diff
@@ -0,0 +1,159 @@
+// limitations under the License.
+package main
+
+import (
+	"fmt"
+	"net"
+	"net/rpc"
+	"net/rpc/jsonrpc"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+
+	"github.com/control-center/serviced/domain/host"
+	"github.com/control-center/serviced/rpc/agent"
+	"github.com/control-center/serviced/utils"
+
+	"github.com/zenoss/glog"
+)
+
+type daemon struct {
+	hostConfig *HostConfig
+	rpcServer  *rpc.Server
+	host       *host.Host
+}
+
+func newDaemon(hostConfig *HostConfig, rpcServer *rpc.Server) (*daemon, error) {
+	d := &daemon{
+		hostConfig: hostConfig,
+		rpcServer:  rpcServer,
+	}
+	return d, nil
+}
+
+func (d *daemon) run() (err error) {
+	if err := d.hostConfig.setDefaults(); err != nil {
+		return fmt.Errorf("unable to initialize host configuration: %v", err)
+	}
+
+	if err := d.buildHost(); err != nil {
+		return fmt.Errorf("failed to build Host domain object: %v", err)
+	}
+
+	d.startRPC()
+	if err := d.registerAgentRPC(); err != nil {
+		glog.Fatal(err)
+	}
+
+	signalC := make(chan os.Signal, 10)
+	signal.Notify(signalC, syscall.SIGINT, syscall.SIGTERM, syscall.SIGHUP)
+	sig := <-signalC
+	glog.Infof("Shutting down due to interrupt - %v", sig)
+	return nil
+}
+
+func (d *daemon) startRPC() {
+	listener, err := net.Listen("tcp", d.hostConfig.Listen)
+	if err != nil {
+		glog.Fatalf("Unable to bind to port %s. Is another instance running?", d.hostConfig.Listen)
+	}
+
+	d.rpcServer.HandleHTTP(rpc.DefaultRPCPath, rpc.DefaultDebugPath)
+
+	glog.Infof("Listening on %s", listener.Addr().String())
+	go func() {
+		for {
+			glog.Infof("listening:Accept()")
+			conn, err := listener.Accept()
+			if err != nil {
+				glog.Fatalf("Error accepting connections: %s", err)
+			}
+			glog.Infof("listening:ServeCodec()")
+			go d.rpcServer.ServeCodec(jsonrpc.NewServerCodec(conn))
+			// err = d.rpcServer.ServeRequest(jsonrpc.NewServerCodec(conn))
+			// if err != nil {
+			// 	glog.Fatalf("Error serving request: %s", err)
+			// }
+		}
+	}()
+}
+
+type MockServer struct {
+	mockHost *host.Host
+}
+
+func (d *daemon) newMock() *MockServer {
+	return &MockServer{
+		mockHost: d.host,
+	}
+}
+
+func (m *MockServer) BuildHost(request agent.BuildHostRequest, hostResponse *host.Host) error {
+	*hostResponse = host.Host{}
+
+	glog.Infof("Build Host Request: %s:%i, %s, %s", request.IP, request.Port, request.PoolID, request.Memory)
+
+	if mem, err := utils.ParseEngineeringNotation(request.Memory); err == nil {
+		m.mockHost.RAMCommitment = mem
+	} else if mem, err := utils.ParsePercentage(request.Memory, m.mockHost.Memory); err == nil {
+		m.mockHost.RAMCommitment = mem
+	} else {
+		return fmt.Errorf("Could not parse RAM Commitment: %v", err)
+	}
+	if request.PoolID != m.mockHost.PoolID {
+		m.mockHost.PoolID = request.PoolID
+	}
+	*hostResponse = *m.mockHost
+	return nil
+	
+}
+
+func (d *daemon) registerAgentRPC() error {
+	// FIXME: We need to register a custom server to implement our own BuildHost() RPC which can return
```

Comment:
You can remove this fix comment because your addition of `d.newMock()` resolves the issue.